src/Components/Facility/FacilityCreate.tsx

```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
-            features: res.data.features || [],
-            ward: res.data.ward_object ? res.data.ward_object.id : 0,
-            kasp_empanelled: res.data.kasp_empanelled
-              ? String(res.data.kasp_empanelled)
-              : "false",
-            address: res.data.address,
-            pincode: res.data.pincode,
-            phone_number:
-              res.data.phone_number.length == 10
-                ? "+91" + res.data.phone_number
-                : res.data.phone_number,
-            latitude: res.data.latitude || "",
-            longitude: res.data.longitude || "",
-            type_b_cylinders: res.data.type_b_cylinders,
-            type_c_cylinders: res.data.type_c_cylinders,
-            type_d_cylinders: res.data.type_d_cylinders,
-            expected_type_b_cylinders: res.data.expected_type_b_cylinders,
-            expected_type_c_cylinders: res.data.expected_type_c_cylinders,
-            expected_type_d_cylinders: res.data.expected_type_d_cylinders,
-            expected_oxygen_requirement: res.data.expected_oxygen_requirement,
-            oxygen_capacity: res.data.oxygen_capacity,
+            facility_type: data.facility_type ? data.facility_type : "",
+            name: data.name ? data.name : "",
+            state: data.state ? data.state : 0,
+            district: data.district ? data.district : 0,
+            local_body: data.local_body ? data.local_body : 0,
+            features: data.features || [],
+            ward: data.ward_object ? data.ward_object.id : 0,
+            kasp_empanelled: "",
+            address: data.address ? data.address : "",
+            pincode: data.pincode ? data.pincode : "",
+            phone_number: data.phone_number
+              ? data.phone_number.length == 10
+                ? "+91" + data.phone_number
+                : data.phone_number
+              : "",
+            latitude: data.location ? String(data.location.latitude) : "",
+            longitude: data.location
+              ? String(data.location.latitude)
+              : "" || "",
+            type_b_cylinders: data.type_b_cylinders,
+            type_c_cylinders: data.type_c_cylinders,
+            type_d_cylinders: data.type_d_cylinders,
+            expected_type_b_cylinders: data.expected_type_b_cylinders,
+            expected_type_c_cylinders: data.expected_type_c_cylinders,
+            expected_type_d_cylinders: data.expected_type_d_cylinders,
+            expected_oxygen_requirement: data.expected_oxygen_requirement,
+            oxygen_capacity: data.oxygen_capacity,
           };
           dispatch({ type: "set_form", form: formData });
-          Promise.all([
-            fetchDistricts(res.data.state),
-            fetchLocalBody(res.data.district),
-            fetchWards(res.data.local_body),
-          ]);
+          setStateId(data.state || 0);
+          setDistrictId(data.district || 0);
+          setLocalBodyId(data.local_body || 0);
```

Comment:
You won't need to do this coalescing if you had undefined/null as uninitialized value

```suggestion
          setStateId(data.state);
          setDistrictId(data.district);
          setLocalBodyId(data.local_body);
```